README.md

```diff
@@ -1,289 +1,610 @@
-
-Extra binding properties **SHOULD** also be defined by the bindable service, using one of the patterns defined in [Pointer to binding data](#pointer-to-binding-data).
-
-The data that is injected or mounted into the container may have a different name because of a few reasons:
-* the backing service may have chosen a different name (discouraged, but allowed).
-* a custom name may have been chosen using the `dataMappings` portion of the `ServiceBinding` CR.
-* a prefix may have been added to certain items in `ServiceBinding`, either globally or per service.  
-
-### Request service binding
-
-Binding is requested by the consuming application, or an entity on its behalf such as the [Runtime Component Operator](https://github.com/application-stacks/runtime-component-operator), via a custom resource that is applied in the same cluster where an implementation of this specification resides.
-
-Since the reference implementation for most of this specification is the [Service Binding Operator](https://github.com/redhat-developer/service-binding-operator) we will be using the `ServiceBinding` CRD, which resides in [this folder](https://github.com/redhat-developer/service-binding-operator/tree/master/deploy/crds), as the entity that holds the binding request.  
-
-**Note** - a few updates / enhancements are being proposed to the current `ServiceBinding` CR, tracked [here](https://github.com/application-stacks/service-binding-specification/issues/16#issuecomment-605309629).
-
-Sample CR
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── type
+│   ├── connection-count
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+    ├── type
+    ├── connection-count
+    ├── uri
+    ├── certificates
+    └── private-key
 ```
+
+# Service Binding
+
+A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.
+
+Restricting service binding to resources within the same namespace is strongly **RECOMMENDED**.  Cross-namespace service binding **SHOULD** be secured appropriately by the implementor to prevent attacks like privilege escalation and secret enumeration.
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference` to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference` to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into the application.
```

Comment:
One of the key things we will want/need to support arbitrary PodSpec-able resources in bindings is an aggregated cluster role.  We want folks to be able to define new PodSpecable resources (e.g. Knative Service/Configuration), which surface a PodSpec, but which the Binding may not know about in advanced.

Similar to the Kubernetes "namespace {admin,editor,viewer}" roles, this would be a standard aggregated cluster role that new types could participate in by "contributing" access to their types via a simple ClusterRole.

Here is an example of a contribution to an Aggregated ClusterRole: https://github.com/knative/serving/blob/6e6e43cb39d7fab108d15f0b10e2c886619cde1b/config/core/roles/clusterrole-namespaced.yaml#L20

Here is an example of a (different) aggregated cluster role definition: https://github.com/knative/serving/blob/6e6e43cb39d7fab108d15f0b10e2c886619cde1b/config/core/200-serviceaccount.yaml#L29-L32

Reply:
Good suggestion. Initially I was thinking RBAC should be in the realm of the implementation, however, defining a ClusterRole label will make it much easier for other CRD providers to expose themselves for binding.

Reply:
Great topic!   I wonder if we add the security related addition as a subsequent PR?

Reply:
On the list for discussion.  https://github.com/application-stacks/service-binding-specification/issues/50